File: services/MealRecipeService/src/Models/Meal.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using MealRecipeService.Models;

namespace MealRecipeService.Models;

[Table("meals")]
public class Meal
{
    [Key]
    [Column("meal_id")]
    public Guid Id { get; set; }

    [Column("name")]
    public string Name { get; set; } = null!;
    [Column("description")]
    public string? Description { get; set; }
    [Column("notes")]
    public string? Notes { get; set; }
    [Column("meal_type_id")]
    public int MealTypeId { get; set; }
    [Column("is_multi_day_meal")]
    public bool IsMultiDayMeal { get; set; }
    [Column("created_by")]
    public Guid CreatedBy {get;set;}
    [Column("created_at")]
    public DateTime CreatedAt { get; set; }
    [Column("updated_by")]
    public Guid UpdatedBy {get;set;}
    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; }
    [Column("owner_user_id")]
    public Guid OwnerUserId { get; set; }

    public MealType MealType { get; set; } = null!;
}

File: frontend/src/types/meal.ts
import type { MealType, ItemType } from './shared'

export interface Meal {
  id: number
  name: string | null
  description: string | null
  notes: string | null
  mealType: MealType
  isMultiDayMeal: boolean
  ownerUserId: number
  createdAt: string
  updatedAt: string
}

export interface MealItem {
  id: number
  mealId: number
  name: string | null
  recipeId: number | null
  itemType: ItemType
  createdAt: string
  updatedAt: string
}

export interface CreateMeal {
  name?: string
  description?: string
  notes?: string
  mealType: MealType
  isMultiDayMeal?: boolean
}

export interface UpdateMeal extends CreateMeal {
  id: number
}

export interface CreateMealItem {
  name?: string
  recipeId?: number
  itemType: ItemType
}

export interface UpdateMealItem extends CreateMealItem {
  id: number
}
